### ISSUE TITLE
Library update hanging on Monterey

### ISSUE THREAD

**Issue by Abinthur** [vdesabou/alfred-spotify-mini-player#534]
I am really sorry but still stuck in this process. Tried re-authenticate ID, reinstall workflow, reopen alfred, but none of them work. A debug email with a zip already sent to you.

Thanks.

_Originally posted by @vdesabou in https://github.com/vdesabou/alfred-spotify-mini-player/issues/533#issuecomment-953891043_

**Comment by Abinthur** [comment]
![Snipaste_2021-10-30_19-52-01](https://user-images.githubusercontent.com/62495250/139531786-389a656f-d659-4fed-a735-d7ac0227d2b9.jpg)
Should be sent. lol

**Comment by Abinthur** [comment]
btw, the favicon now is odd too.
![Snipaste_2021-10-30_19-57-20](https://user-images.githubusercontent.com/62495250/139531971-2abe57e0-b648-429a-97d6-9e123a10e1eb.jpg)
(now i can only get notifications of currently listening music with 0 music in library...)

**Comment by vdesabou** [comment]
> btw, the favicon now is odd too.
> ![Snipaste_2021-10-30_19-57-20](https://user-images.githubusercontent.com/62495250/139531971-2abe57e0-b648-429a-97d6-9e123a10e1eb.jpg)
> (now i can only get notifications of currently listening music with 0 music in library...)

You can run that 
![CleanShot 2021-11-02 at 10 37 01](https://user-images.githubusercontent.com/4061923/139822210-c02f67ec-de78-4c1a-b808-c558000705f1.jpg)

**Comment by vdesabou** [comment]
> Should be sent. lol

It was in my SPAM folder yes

**Comment by jcmarinn** [comment]
I was having same problem (stuck on initializing) but after install I noticed it gave me error that it could not execute the "Spotify Mini Player" app since developer was uknown and app downloaded from Internet

I deleted app, re-installed workflow and before updating library went to location of Spotify Mini Player App (within Alfred.prefereces package/workflows .........) and opened form finder and told to open even if uknown and then launched the update on library and it worked this time

**Comment by vdesabou** [comment]
@jcmarinn Thanks for the feedback. I'll call fix permissions automatically, I'll include that in next version

**Comment by vdesabou** [comment]
a user reported the error below, which might explain the problem seen here:

```
[23:42:40.429] STDERR: Spotify Mini Player[Run Script] PHP Warning:  file_get_contents(/src/public.key): Failed to open stream: No such file or directory in /Users/Cambridge/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.6A29746C-17C0-470A-B2BB-1D53DF31D79F/src/functions.php on line 2710
PHP Warning:  openssl_public_encrypt(): key parameter is not a valid public key in /Users/Cambridge/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.6A29746C-17C0-470A-B2BB-1D53DF31D79F/src/functions.php on line 2713
PHP Warning:  file_get_contents(/src/public.key): Failed to open stream: No such file or directory in /Users/Cambridge/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.6A29746C-17C0-470A-B2BB-1D53DF31D79F/src/functions.php on line 2710
PHP Warning:  openssl_public_encrypt(): key parameter is not a valid public key in /Users/Cambridge/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.6A29746C-17C0-470A-B2BB-1D53DF31D79F/src/functions.php on line 2713
cat: /tmp/spot_mini_debug/cache/spotify_mini_player_web_server.log: No such file or directory


[23:45:55.195] ERROR: Spotify Mini Player[Run Script] PHP Warning:  Undefined variable $savedMySavedShows in /Users/Cambridge/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.6A29746C-17C0-470A-B2BB-1D53DF31D79F/src/refreshLibrary.php on line 545
PHP Fatal error:  Uncaught TypeError: sizeof(): Argument #1 ($value) must be of type Countable|array, null given in /Users/Cambridge/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.6A29746C-17C0-470A-B2BB-1D53DF31D79F/src/refreshLibrary.php:545
Stack trace:
#0 /Users/Cambridge/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.6A29746C-17C0-470A-B2BB-1D53DF31D79F/src/action.php(2175): refreshLibrary(Object(Workflows))
#1 /Users/Cambridge/Library/Caches/com.runningwithcrayons.Alfred/Workflow Scripts/9090D581-AF59-49B6-8BF6-0A5FA7CC0364(4): main('a:16:{i:0;s:0:"...', 'TRACK', '')
#2 {main}
  thrown in /Users/Cambridge/Library/Application Support/Alfred/Alfred.alfredpreferences/workflows/user.workflow.6A29746C-17C0-470A-B2BB-1D53DF31D79F/src/refreshLibrary.php on line 545
```

**Comment by vdesabou** [comment]
Is this still not working for people using Monterey and 11.6 ? If so please follow those steps https://github.com/vdesabou/alfred-spotify-mini-player/issues/534#issuecomment-954092447

**Comment by Abinthur** [comment]
still not working for me. Hope you found solutions with more feedbacks.

**Comment by vdesabou** [comment]
> still not working for me. Hope you found solutions with more feedbacks.

Can you update to latest and follow this https://github.com/vdesabou/alfred-spotify-mini-player/issues/534#issuecomment-954092447

can you also describe exactly what you see or get? 

There are a lot of users for which it is working on Monterey, so it does not seem a generic issue

**Comment by Abinthur** [comment]
Almost burst into tears to see it work on Monterey for the first time!!!
thanks!!!

**Comment by vdesabou** [comment]
> Almost burst into tears to see it work on Monterey for the first time!!!
> thanks!!!

Ok 👌 i suspect you had the issue which was fixed in 11.6

i will close that ticket then
